src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -521,23 +521,23 @@ private String key() {
                 boolean needsToPersist = completed == null;
                 super.onLoad();
 
-                if (completed == Boolean.TRUE && result == null) {
+                if (Boolean.TRUE.equals(completed) && result == null) {
                     LOGGER.log(Level.FINE, "Completed build with no result set, defaulting to failure for "+this);
                     setResult(Result.FAILURE);
                     needsToPersist = true;
                 }
 
                 // TODO See if we can simplify this, especially around interactions with 'completed'.
 
-                if (execution != null && completed != Boolean.TRUE) {
+                if (execution != null && !Boolean.TRUE.equals(completed)) {
                     FlowExecution fetchedExecution = getExecution();  // Triggers execution.onLoad so we can resume running if not done
 
                     if (fetchedExecution != null) {
                         if (completed == null) {
                             completed = Boolean.valueOf(fetchedExecution.isComplete());
                         }
 
-                        if (!completed == Boolean.TRUE) {
+                        if (Boolean.FALSE.equals(completed)) {
```

Comment:
Unrelated to the other changes. The old code was fine, since we check `completed == null` a few lines up and set it to a non-null value, but unnecessarily roundabout, since it was unboxing the `Boolean` to a `boolean`, then inverting that and comparing the inverted result to `Boolean.TRUE`, which was also unboxed into a `boolean`. Much clearer in my opinion to just compare to `Boolean.FALSE` directly.